showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java

```diff
@@ -0,0 +1,138 @@
+
+/**
+ * Test suite to confirm a client can be instantiated with GDCH credentials. No calls are made since
+ * it is not feasible to test against real GDCH servers (or replicate an environment)
+ */
+public class ITGdch {
+
+  private static final String TEST_GDCH_CREDENTIAL_FILE = "/test_gdch_credential.json";
+  private static final String CA_CERT_RESOURCE_PATH = "/fake_cert.pem";
+  private static final String CA_CERT_JSON_KEY = "ca_cert_path";
+  private static final String TEMP_CREDENTIAL_JSON_FILENAME = "temp_gdch_credential.json";
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  private EchoClient client;
+  private EchoSettings settings;
+  private Credentials credentials;
+
+  @Before
+  public void setup() throws IOException, URISyntaxException {
+    prepareCredentials();
+    tempFolder.create();
+    settings =
+        EchoSettings.newBuilder()
+            .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+            .build();
+  }
+
+  @After
+  public void tearDown() {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  private void prepareCredentials() throws IOException, URISyntaxException {
+    // compute absolute path of the CA certificate
+    Path caCertPath = Paths.get(getClass().getResource(CA_CERT_RESOURCE_PATH).toURI());
+
+    // open gdch credential json (still needs its "ca_cert_path" to point to the CA certificate
+    // obtained from above)
+    JsonFactory factory = new GsonFactory();
+    GenericJson converted =
+        factory.fromInputStream(
+            getClass().getResourceAsStream(TEST_GDCH_CREDENTIAL_FILE), GenericJson.class);
+
+    // modify and save to a temporary folder
+    converted.set(CA_CERT_JSON_KEY, caCertPath.toAbsolutePath().toString());
+    File tempGdchCredentialFile = tempFolder.newFile(TEMP_CREDENTIAL_JSON_FILENAME);
+    try (FileWriter fileWriter = new FileWriter(tempGdchCredentialFile)) {
+      String preparedJson = converted.toPrettyString();
+      fileWriter.write(preparedJson);
+    }
+
+    // use temp location to instantiate credentials
+    credentials = GdchCredentials.fromStream(new FileInputStream(tempGdchCredentialFile));
+  }
+
+  /**
+   * {@link com.google.api.gax.rpc.ClientContext} sets a credentials object many times before
+   * concluding its creation. double check that they end up being the same
+   */
+  @Test
+  public void testClientWithGdchCredential_keepsCredentials() throws IOException {
+    client = EchoClient.create(settings);
+    assertSame(credentials, client.getSettings().getCredentialsProvider().getCredentials());
+  }
+
+  /**
+   * Confirms setting a valid audience is successful.
+   *
+   * <p>The provider's credentials do not get altered when adding audience. They are recreated and
+   * used only inside `ClientContext`
+   *
+   * @throws IOException
+   * @throws URISyntaxException
+   */
+  @Test
+  public void testClientWithGdchCredentialWithValidAudience_correct()
```

Comment:
Can we have an integration test that trigger refresh and check that aud exists in the request via fake transport? Probably other test cases could be validated similarly via refresh as well

Reply:
I just modified the tests to confirm audience works by refreshing the token. I imported some test classes from `google-auth-library`. I had to propose a PR for this https://github.com/googleapis/google-auth-library-java/pull/1224